packager/packager.cc

```diff
@@ -681,6 +688,80 @@ class PackageInfoImpl final : public core::packages::PackageInfo {
         // No path found
         return nullopt;
     }
+
+    void trackMissingImport(const core::FileRef file, const core::packages::MangledName toImport,
+                            const core::packages::ImportType importType,
+                            const core::AutocorrectSuggestion autocorrect) {
+        auto r = trackedMissingImports_.find({toImport, importType});
+        if (r != trackedMissingImports_.end()) {
+            r->second.first.insert(file);
+        } else {
+            trackedMissingImports_[{toImport, importType}] = {{file}, autocorrect};
+        }
+    }
+
+    void untrackMissingImportsFor(const core::FileRef file) {
+        for (auto &[_, value] : trackedMissingImports_) {
+            auto &files = value.first;
+            if (files.contains(file)) {
+                files.erase(file);
+            }
```

Comment:
Nit: better to do something like:

```c++
auto it = files.find(file);
if (it == files.end()) {
  continue;
}
files.erase(it);
```

so you don't do unnecessary hash lookups.